presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -965,10 +965,12 @@ private Table prepareTable(ConnectorSession session, ConnectorTableMetadata tabl
 
         validateColumns(hiveStorageFormat, columnHandles);
 
+        MetastoreContext metastoreContext = new MetastoreContext(session.getIdentity(), session.getQueryId(), session.getClientInfo(), session.getSource(), getMetastoreHeaders(session), isUserDefinedTypeEncodingEnabled(session), columnConverterProvider);
```

Comment:
I found it would be helpful to create a helper for this function in this class...
The helper only needs to take `session` and `columnConverterProvider` as the inputs. So we can avoid repeating such long constructors everywhere in the class.